source/extensions/tracers/xray/localized_sampling.cc

```diff
@@ -0,0 +1,185 @@
+#include "extensions/tracers/xray/localized_sampling.h"
+
+#include "common/http/exception.h"
+
+#include "extensions/tracers/xray/util.h"
+
+#include "rapidjson/document.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Tracers {
+namespace XRay {
+
+constexpr static double DefaultRate = 0.5;
```

Comment:
I may have mentioned before, but a ConstSingleton (Envoy thing) is usually used to store constants. But I do prefer this, especially since they are just static json keys. I think this will get through fine as is.